src/bdev_desc.rs

```diff
@@ -375,3 +293,112 @@ impl<T: BdevOps> LockContext<T> {
 pub struct LbaRangeLock<T: BdevOps> {
     ctx: Box<LockContext<T>>,
 }
+
+/// Allows gaining exclusive access over a block range.
+/// # Warning
+///     Due to the lack of async drop the range lock must be
```

Comment:
I wonder if we can still release in the drop method, just not wait for it - and then the caller may decide to either wait or just allow the drop to do it if it doesn't care about waiting for unlock to complete, wdyt @dsavitskiy ?

Reply:
As I understand this drop part - since we have nothing custom to do in Drop, and the allocated heap memory will be anyway freed by default drop, it seems ok to wait for unlock to complete.